Python/segment_tree.py

```diff
@@ -0,0 +1,156 @@
+import math
+
+
+class SegmentTree:
+    '''Class representation of Segment Tree
+    
+    ...
+
+    Attributes:
+    ----------
+    arr: Any iterable
+        the segment tree will be built using this iterable
+    len-arr: int
+        length of the iterable arr
+    func: method/function
+        the function that will be used to built the tree
+    default: Any
+        the default value with which the tree array is initialized
+    tree: list
+        the representation of the tree
+
+    Note: Indexing is 0 based in this implementation
+    '''
+    def __init__(self, arr, func, default):
+
+        # Checking if iterable passed is of NoneType
+        if arr is not None:
+            self.arr = arr
+            self.len_arr = len(self.arr)
+        else:
+            raise TypeError('Argument arr should not be null')
+
+        # Checking if function passed is of NoneType
+        if func is not None:
+            self.func = func
+        else:
+            raise TypeError('Argument func should not be null')
+
+        self.default = default
+        self._build_tree()
+
+    def _is_power_of_2(self, n):
```

Comment:
Instead of using heavy library functions. It would be appreciated if you use bitwise operations to do the same.
```
def _is_power_of_2(n):
    return (n & n - 1) == 0
```

Reply:
Thank you for this suggestion.
Appreciated!